cmd/ocsp-updater/main.go

```diff
@@ -129,11 +140,63 @@ func newUpdater(
 		})
 	}
 
-	updater.ocspMinTimeToExpiry = config.OCSPMinTimeToExpiry.Duration
+	if config.AkamaiBaseURL != "" {
+		issuer, err := core.LoadCert(issuerPath)
+		ccu, err := akamai.NewCachePurgeClient(
+			config.AkamaiBaseURL,
+			config.AkamaiClientToken,
+			config.AkamaiClientSecret,
+			config.AkamaiAccessToken,
+			config.AkamaiPurgeRetries,
+			config.AkamaiPurgeRetryBackoff.Duration,
+			log,
+			stats,
+		)
+		if err != nil {
+			return nil, err
+		}
+		updater.ccu = ccu
+		updater.issuer = issuer
+	}
 
 	return &updater, nil
 }
 
+// sendPurge should only be called as a Goroutine as it will block until the purge
+// request is successful
+func (updater *OCSPUpdater) sendPurge(der []byte) {
+	cert, err := x509.ParseCertificate(der)
+	if err != nil {
+		updater.log.AuditErr(fmt.Errorf("Failed to parse certificate for cache purge"))
+		return
+	}
+
+	req, err := ocsp.CreateRequest(cert, updater.issuer, nil)
+	if err != nil {
+		updater.log.AuditErr(fmt.Errorf("Failed to create OCSP request for cache purge"))
+		return
+	}
+
+	// Create a GET style OCSP url for each endpoint in cert.OCSPServer (still waiting
+	// on word from Akamai on how to properly purge cached POST requests, for now just
+	// do GET)
+	urls := []string{}
+	for _, ocspServer := range cert.OCSPServer {
+		if !strings.HasSuffix(ocspServer, "/") {
+			ocspServer += "/"
+		}
+		urls = append(
+			urls,
+			fmt.Sprintf("%s%s", ocspServer, url.QueryEscape(base64.StdEncoding.EncodeToString(req))),
```

Comment:
this can be just `ocspServer + url.QueryEscape(...)`

Also, you can use https://golang.org/pkg/path/#Join instead of doing the Suffix check.